### ISSUE TITLE
Controller for Mongodb Sub-Documents

### ISSUE THREAD

**Issue by ghost** [balderdashy/sails#1250]
## The Problem

I have a model that looks like that:

```
// Person.js

var Board = {
  attributes: {
   name: "STRING"
  }
}

var Person = {
  attributes: {
    firstName: 'STRING',
    lastName: 'STRING',
    boards: [Board],
    }
  }
};

module.exports = Person;
```

The model `Person` contains a subdocument called `Boards` that is essentially, an array of `Board` models.

I'd like to have routes that would allow me to access either all boards owned by a user, or a single board. The routes should look like that:

```
GET /Person/Boards/
GET /Person/Boards/:id
```
## Potential Solution

Currently, the only way I can see this happening, is by having my own `actions` inside of my controller:

```
  boards: (req, res, next) ->
    // Find the current user from the db, and req.send his boards
  boards/:id: (req, res, next) ->
   // Find the current user from the db, and req.send board where boards.id == req.params.id
```

Basically, I'm re-building a REST api for this sub-document.

Is there any other, better/automated way to do so? :)
